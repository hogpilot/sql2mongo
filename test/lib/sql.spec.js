const { getSqlAst, parseSelect } = require("../../lib/sql");

describe("sql", () => {
  describe("getSqlAst", () => {
    it("Throws an error if query is invalid", () => {
      const invalidQuery = "SELECT FROM someTable";

      expect(() => {
        getSqlAst(invalidQuery);
      }).toThrow();
    });

    it("Returns the AST if query is valid", () => {
      const validQuery = "SELECT * FROM someTable";

      const result = getSqlAst(validQuery);

      expect(result).toMatchSnapshot();
    });
  });

  describe("parseSelect", () => {
    it("returns an empty object if Asterisk is passed", () => {
      const asteriskSelectValue = [{ value: "*", type: "Identifier" }];

      const result = parseSelect(asteriskSelectValue);

      expect(result).toMatchObject({});
    });

    it("returns a valid object if fields are passed", () => {
      const fieldsSelectValue = [
        { value: "name", type: "Identifier" },
        { value: "age", type: "Identifier" },
        { value: "city", type: "Identifier" }
      ];

      const result = parseSelect(fieldsSelectValue);

      expect(result).toMatchObject({ name: 1, age: 1, city: 1 });
    });
  });
});
