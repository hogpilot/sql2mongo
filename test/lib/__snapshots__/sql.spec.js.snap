// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sql getSqlAst Returns the AST if query is valid 1`] = `
Object {
  "hasSemicolon": false,
  "nodeType": "Main",
  "value": Object {
    "distinctOpt": null,
    "from": Object {
      "type": "TableReferences",
      "value": Array [
        Object {
          "type": "TableReference",
          "value": Object {
            "alias": null,
            "hasAs": null,
            "indexHintOpt": null,
            "partition": null,
            "type": "TableFactor",
            "value": Object {
              "type": "Identifier",
              "value": "someTable",
            },
          },
        },
      ],
    },
    "groupBy": null,
    "having": null,
    "highPriorityOpt": null,
    "limit": null,
    "maxStateMentTimeOpt": null,
    "orderBy": null,
    "partition": null,
    "procedure": null,
    "selectItems": Object {
      "type": "SelectExpr",
      "value": Array [
        Object {
          "type": "Identifier",
          "value": "*",
        },
      ],
    },
    "sqlBigResultOpt": null,
    "sqlBufferResultOpt": null,
    "sqlCacheOpt": null,
    "sqlCalcFoundRowsOpt": null,
    "sqlSmallResultOpt": null,
    "straightJoinOpt": null,
    "type": "Select",
    "updateLockMode": null,
    "where": null,
  },
}
`;
